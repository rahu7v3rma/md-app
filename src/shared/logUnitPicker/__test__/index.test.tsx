import React from 'react';
import renderer, { act } from 'react-test-renderer';

import LogUnitPicker from '@/shared/logUnitPicker';

it('logunitpicker snapshot', async () => {
    const tree = renderer.create(<LogUnitPicker value={1} />);
    expect(tree).toMatchSnapshot();
});

it('logunitpicker renders without crashing', async () => {
    const tree = renderer.create(<LogUnitPicker value={1} />);
    expect(tree.root).toBeTruthy();
});

it('displays the unit value correctly', async () => {
    const tree = renderer.create(
        <LogUnitPicker value={1} valueTextTestID={'valueTextTestID'} />
    );
    const valueTextTestID = tree.root.findByProps({
        testID: 'valueTextTestID'
    }).props;
    expect(valueTextTestID.children).toBe(1);
});

it('calls onIncrementHandler when the plus button is pressed', async () => {
    const onIncrementHandler = jest.fn();
    const tree = renderer.create(
        <LogUnitPicker
            value={1}
            onIncrementHandler={onIncrementHandler}
            plusButtonTestID={'plusButtonnTestID'}
        />
    );
    const plusButton = tree.root.findByProps({
        testID: 'plusButtonnTestID'
    }).props;
    await act(() => plusButton.onPress());
    expect(onIncrementHandler).toHaveBeenCalledTimes(1);
});

it('calls onDecrementHandler when the minus button is pressed', async () => {
    const onDecrementHandler = jest.fn();
    const tree = renderer.create(
        <LogUnitPicker
            value={1}
            onDecrementHandler={onDecrementHandler}
            minusButtonTestID={'minusButtonTestID'}
        />
    );
    const minusButton = tree.root.findByProps({
        testID: 'minusButtonTestID'
    }).props;
    await act(() => minusButton.onPress());
    expect(onDecrementHandler).toHaveBeenCalledTimes(1);
});

it('calls onChangeHandler when the unit value is edited', async () => {
    const onChangeHandler = jest.fn();
    const tree = renderer.create(
        <LogUnitPicker
            value={1}
            onChangeHandler={onChangeHandler}
            touchValueTestID={'touchValueTestID'}
            textInputTestID={'textInputTestID'}
        />
    );
    // make input editable first
    const touchValue = tree.root.findByProps({
        testID: 'touchValueTestID'
    }).props;
    await act(() => touchValue.onPress());

    // now we can edit value
    const textInput = tree.root.findByProps({
        testID: 'textInputTestID'
    }).props;
    await act(() => textInput.onChangeText('12'));
    expect(onChangeHandler).toHaveBeenCalledWith(12);
});

it('displays the timer value correctly', async () => {
    const tree = renderer.create(
        <LogUnitPicker
            value={90}
            type={'timer'}
            hourTextTestID={'hourTextTestID'}
            minTextTestID={'minTextTestID'}
        />
    );
    // call useEffect method
    await act(async () => {
        renderer.create(
            <LogUnitPicker
                value={90}
                type={'timer'}
                hourTextTestID={'hourTextTestID'}
                minTextTestID={'minTextTestID'}
            />
        );
    });
    const hourText = tree.root.findByProps({
        testID: 'hourTextTestID'
    }).props;
    expect(hourText.children).toBe('1');
    const minTex = tree.root.findByProps({
        testID: 'minTextTestID'
    }).props;
    expect(minTex.children).toBe('30');
});

it('opens the timer picker modal when timer is pressed', async () => {
    const tree = renderer.create(
        <LogUnitPicker
            value={1}
            type={'timer'}
            timerTouchTestID={'timerTouchTestID'}
            timePicModalTestID={'timePicModalTestID'}
        />
    );
    // clickon timer to open modal
    const timerTouch = tree.root.findByProps({
        testID: 'timerTouchTestID'
    }).props;
    await act(() => timerTouch.onPress());

    // check modal is visible or not
    const timePicModal = tree.root.findByProps({
        testID: 'timePicModalTestID'
    }).props;
    expect(timePicModal).toBeTruthy();
});
