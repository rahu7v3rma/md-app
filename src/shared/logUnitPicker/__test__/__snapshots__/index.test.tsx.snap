// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logunitpicker snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginBottom": 15,
      "marginTop": 30,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={64}
      bbWidth={64}
      fill="none"
      focusable={false}
      height={64}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 64,
            "width": 64,
          },
        ]
      }
      vbHeight={64}
      vbWidth={64}
      width={64}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGGroup
          matrix={
            [
              1,
              0,
              0,
              -1,
              16,
              48,
            ]
          }
        >
          <RNSVGCircle
            cx={16}
            cy={16}
            fill={4294967295}
            propList={
              [
                "fill",
              ]
            }
            r={16}
          />
          <RNSVGCircle
            cx={16}
            cy={16}
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            r={14}
            stroke={4283256653}
            strokeWidth={4}
          />
        </RNSVGGroup>
        <RNSVGRect
          fill={4283256653}
          height={1.66667}
          propList={
            [
              "fill",
              "stroke",
            ]
          }
          rx={0.833333}
          stroke={4283256653}
          width={13.3333}
          x={25.3333}
          y={30.8322}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 20,
        "justifyContent": "center",
        "paddingVertical": 20,
        "width": 214,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        style={
          [
            {
              "fontFamily": "Poppins",
            },
            undefined,
            {
              "color": "#1F1E22",
              "fontSize": 32,
              "fontWeight": "bold",
              "lineHeight": 40,
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={64}
      bbWidth={64}
      fill="none"
      focusable={false}
      height={64}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 64,
            "width": 64,
          },
        ]
      }
      vbHeight={64}
      vbWidth={64}
      width={64}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGGroup
          matrix={
            [
              1,
              0,
              0,
              -1,
              16,
              48,
            ]
          }
        >
          <RNSVGCircle
            cx={16}
            cy={16}
            fill={4294967295}
            propList={
              [
                "fill",
              ]
            }
            r={16}
          />
          <RNSVGCircle
            cx={16}
            cy={16}
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            r={14}
            stroke={4283256653}
            strokeWidth={4}
          />
        </RNSVGGroup>
        <RNSVGPath
          d="M26 32h12m-6-6v12"
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={4283256653}
          strokeLinecap={1}
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
